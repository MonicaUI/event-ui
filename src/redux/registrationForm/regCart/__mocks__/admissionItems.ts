/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';
import EventSnapshot from '../../../../../fixtures/EventSnapshot.json';
import appData from '../__tests__/appData.json';
import { REGISTERING } from '../../../registrationIntents';

export const selectAdmissionItem = jest.fn(noopAction);
export const unSelectAdmissionItem = jest.fn(noopAction);
export function RegCartClient(): $TSFixMe {}
export function ProductVisibilityClient(): $TSFixMe {}
export function EventSnapshotClient(): $TSFixMe {}
export const eventId = EventSnapshot.eventSnapshot.id;
export const primaryEventRegId = '00000000-0000-0000-0000-000000000001';
export const regCartId = '928f02df-4ba7-4637-824d-ae71bd414e4f';
export const unSelectedAdmissionId = 'unSelectedAdmissionId';

export const visibleProductsForDefaultRegType = {
  admissionItems: {
    '084ada84-832c-4511-ab20-6727fcf8d5ec': {
      code: '',
      description: '',
      id: '084ada84-832c-4511-ab20-6727fcf8d5ec',
      capacityId: '084ada84-832c-4511-ab20-6727fcf8d5ec',
      name: 'Event Registration',
      status: 2,
      type: 'AdmissionItem',
      defaultFeeId: '00000000-0000-0000-0000-000000000000',
      fees: {},
      closedReasonType: 'NotClosed',
      isOpenForRegistration: true,
      limitOptionalItemsToSelect: false,
      includeWaitlistSessionsTowardsMaximumLimit: false,
      applicableContactTypes: ['00000000-0000-0000-0000-000000000000'],
      limitOptionalSessionsToSelect: false,
      associatedOptionalSessions: [],
      applicableOptionalItems: [],
      minimumNumberOfSessionsToSelect: 0,
      availableOptionalSessions: [],
      displayOrder: 2
    }
  },
  sessionProducts: {},
  sessions: null,
  sessionGroups: null,
  sortKeys: {},
  quantityItems: {},
  donationItems: {},
  skipValidationItems: null
};

export function getState(regMod = false): $TSFixMe {
  const regCart = {
    regCartId,
    status: 'INPROGRESS',
    eventSnapshotVersions: {
      [(EventSnapshot as $TSFixMe).id]: 'fake-eventSnapshot-version'
    },
    eventRegistrations: {
      [primaryEventRegId]: {
        eventRegistrationId: primaryEventRegId,
        eventId,
        attendee: {
          personalInformation: {
            contactId: 'd10d9eab-743f-4b3e-80ee-adb520920281',
            emailAddress: 'aa@j.mail',
            firstName: 'a',
            lastName: 'a',
            primaryAddressType: 'WORK'
          },
          attendeeId: 'aac17ae9-74b8-4687-ae9e-e2781d247f95'
        },
        attendeeType: 'ATTENDEE',
        productRegistrations: [
          {
            productId: Object.keys(EventSnapshot.eventSnapshot.products.admissionItems)[0],
            productType: 'AdmissionItem',
            quantity: 1,
            requestedAction: 'REGISTER'
          }
        ],
        quantityItemRegistrations: {},
        sessionRegistrations: {},
        registrationTypeId: '00000000-0000-0000-0000-000000000000',
        registrationPathId: '411c6566-1e5a-4c38-b8e5-f63ab9239b40'
      }
    },
    regMod
  };
  return {
    website: EventSnapshot.eventSnapshot.siteEditor.website,
    accessToken: '',
    clients: {
      // @ts-expect-error ts-migrate(2350) FIXME: Only a void function can be called with the 'new' ... Remove this comment to see the full error message
      regCartClient: new RegCartClient(),
      // @ts-expect-error ts-migrate(2350) FIXME: Only a void function can be called with the 'new' ... Remove this comment to see the full error message
      productVisibilityClient: new ProductVisibilityClient(),
      // @ts-expect-error ts-migrate(2350) FIXME: Only a void function can be called with the 'new' ... Remove this comment to see the full error message
      eventSnapshotClient: new EventSnapshotClient()
    },
    registrationForm: {
      regCartPayment: {
        pricingInfo: {
          selectedPaymentMethod: null
        }
      },
      regCart,
      errors: {}
    },
    regCartStatus: {
      registrationIntent: REGISTERING,
      lastSavedRegCart: regCart
    },
    event: {
      eventFeatureSetup: {
        fees: {
          fees: false
        },
        registrationProcess: {
          multipleRegistrationTypes: false
        },
        agendaItems: {
          admissionItems: true
        }
      },
      products: {}
    },
    visibleProducts: {
      Sessions: {
        [primaryEventRegId]: { ...visibleProductsForDefaultRegType }
      }
    },
    appData,
    text: {
      translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx),
      resolver: {
        date: () => 'some date',
        currency: x => x
      }
    },
    userSession: {
      regCartId: '928f02df-4ba7-4637-824d-ae71bd414e4f'
    },
    defaultUserSession: {},
    travelCart: {}
  };
}

export const response = {
  get regCart(): $TSFixMe {
    return {
      regCartId,
      status: 'INPROGRESS',
      eventRegistrations: {
        [primaryEventRegId]: {
          eventRegistrationId: primaryEventRegId,
          attendee: {
            personalInformation: {
              emailAddress: 'aa@j.mail'
            },
            attendeeId: 'aac17ae9-74b8-4687-ae9e-e2781d247f95'
          },
          productRegistrations: [
            {
              productId: Object.keys(EventSnapshot.eventSnapshot.products.admissionItems)[0],
              productType: 'AdmissionItem',
              quantity: 1,
              requestedAction: 'REGISTER'
            }
          ],
          sessionRegistrations: {},
          registrationTypeId: '00000000-0000-0000-0000-000000000000',
          registrationPathId: '411c6566-1e5a-4c38-b8e5-f63ab9239b40'
        }
      }
    };
  }
};
