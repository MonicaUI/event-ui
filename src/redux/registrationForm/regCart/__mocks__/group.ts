/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';

export const removeEventRegistrationFromRegCart = jest.fn(noopAction);

export const removeGroupMembersFromRegCart = jest.fn(noopAction);

export const addGroupMemberInRegCart = jest.fn(noopAction);

export const navigateToGroupMemberRegistration = jest.fn(noopAction);
