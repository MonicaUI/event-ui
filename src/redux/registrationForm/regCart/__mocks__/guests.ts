/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';

export const updateRegCartWithGuests = jest.fn(regCart => regCart);

export const updateGuestsInRegCart = jest.fn(noopAction);

export const setCurrentGuestEventRegistration = jest.fn(noopAction);

export const removeGuestByEventRegistrationId = jest.fn(regCart => regCart);

export const updateGuestDetails = jest.fn(noopAction);

export const clearTemporaryGuestInformation = jest.fn(noopAction);
