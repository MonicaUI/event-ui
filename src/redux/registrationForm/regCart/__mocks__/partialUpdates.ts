/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';

export const applyPartialEventRegistrationUpdate = jest.fn(noopAction);
