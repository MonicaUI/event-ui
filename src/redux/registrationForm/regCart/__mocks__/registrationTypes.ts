/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';

export const runRegistrationTypeChangeValidationsForPrimaryAndGuest = jest.fn(noopAction);
export const setRegistrationTypeId = jest.fn(noopAction);
