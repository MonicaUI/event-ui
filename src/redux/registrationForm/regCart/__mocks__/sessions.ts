/* eslint-env jest */
import noopAction from '../../../../testUtils/noopAction';
import EventSnapshot from '../../../../../fixtures/EventSnapshot.json';
import appData from '../__tests__/appData.json';
import { REGISTERING } from '../../../registrationIntents';

export const unSelectSession = jest.fn(noopAction);
export const selectSession = jest.fn(noopAction);
export const switchSession = jest.fn(noopAction);

export const registrationPathId = '411c6566-1e5a-4c38-b8e5-f63ab9239b40';
export const regCartId = '928f02df-4ba7-4637-824d-ae71bd414e4f';
export const eventId = EventSnapshot.eventSnapshot.id;
export const accessToken = '';
export const primaryEventRegId = '00000000-0000-0000-0000-000000000001';

export const unSelectedSessionId = 'unSelectedSessionId';
export const waitlistedSessionId = 'waitlistSessionId';
export const selectedSessionId = 'sessionId';
export const nonWaitlistedSessionId = 'nonWaitlistSessionId';

export function RegCartClient(): $TSFixMe {}
export const visibleProductsForRegistrationWithSessionGroups = {
  admissionItems: {
    admissionItemId: {
      id: 'admissionItemId'
    }
  },
  sessionProducts: {
    sessionGroup: {
      code: 'Group',
      id: 'sessionGroup',
      isOpenForRegistration: true,
      name: 'Group',
      sessions: {
        freeSessionInGroup: {
          capacityId: 'unlimitedCapacity',
          id: 'freeSessionInGroup',
          name: 'Free Session in Group'
        },
        sessionInGroupWithFee: {
          capacityId: 'unlimitedCapacity',
          id: 'sessionInGroupWithFee',
          name: 'Session in Group'
        }
      }
    }
  }
};

export const visibleProductsForRegistration = {
  admissionItems: {
    admissionItemId: {
      id: 'admissionItemId'
    }
  },
  sessionProducts: {
    [selectedSessionId]: {
      id: selectedSessionId,
      capacityId: selectedSessionId,
      isWaitlistEnabled: true,
      name: selectedSessionId,
      waitlistCapacityId: `${selectedSessionId}_waitlist`
    },
    [unSelectedSessionId]: {
      id: unSelectedSessionId,
      capacityId: unSelectedSessionId,
      name: unSelectedSessionId,
      isWaitlistEnabled: true,
      waitlistCapacityId: `${unSelectedSessionId}_waitlist`
    },
    limitedSelectedSession: {
      id: 'limitedSelectedSession',
      capacityId: 'limitedSelectedSession',
      isWaitlistEnabled: false,
      name: 'limitedSelectedSession',
      waitlistCapacityId: nonWaitlistedSessionId
    },
    limitedUnSelectedSession: {
      id: 'limitedUnSelectedSession',
      capacityId: 'limitedUnSelectedSession',
      isWaitlistEnabled: false,
      name: 'limitedUnSelectedSession',
      waitlistCapacityId: nonWaitlistedSessionId
    },
    [nonWaitlistedSessionId]: {
      id: nonWaitlistedSessionId,
      capacityId: nonWaitlistedSessionId,
      isWaitlistEnabled: false,
      name: nonWaitlistedSessionId,
      waitlistCapacityId: nonWaitlistedSessionId
    },
    [waitlistedSessionId]: {
      id: waitlistedSessionId,
      capacityId: waitlistedSessionId,
      isWaitlistEnabled: true,
      name: waitlistedSessionId,
      waitlistCapacityId: `${waitlistedSessionId}_waitlist`
    }
  },
  sortKeys: {
    [selectedSessionId]: ['2017-09-10T22:00:00.000Z'],
    [unSelectedSessionId]: ['2017-09-10T22:00:00.000Z'],
    [nonWaitlistedSessionId]: ['2017-09-10T22:00:00.000Z'],
    [waitlistedSessionId]: ['2017-09-10T22:00:00.000Z']
  }
};

export const dummyCapacitySummaries = {
  [selectedSessionId]: {
    capacityId: selectedSessionId,
    totalCapacityAvailable: -1,
    availableCapacity: -1,
    active: true
  },
  [unSelectedSessionId]: {
    capacityId: unSelectedSessionId,
    totalCapacityAvailable: -1,
    availableCapacity: -1,
    active: true
  },
  [waitlistedSessionId]: {
    capacityId: waitlistedSessionId,
    totalCapacityAvailable: -1,
    availableCapacity: -1,
    active: true
  },
  limitedSelectedSession: {
    capacityId: 'limitedSelectedSession',
    totalCapacityAvailable: 10,
    availableCapacity: 10,
    active: true
  },
  limitedUnSelectedSession: {
    capacityId: 'limitedUnSelectedSession',
    totalCapacityAvailable: 20,
    availableCapacity: 10,
    active: true
  }
};

export function getState(): $TSFixMe {
  const regCart = {
    regCartId,
    status: 'INPROGRESS',
    eventSnapshotVersions: {
      [(EventSnapshot as $TSFixMe).id]: 'fake-eventSnapshot-version'
    },
    eventRegistrations: {
      [primaryEventRegId]: {
        eventRegistrationId: primaryEventRegId,
        eventId,
        attendee: {
          personalInformation: {
            contactId: 'd10d9eab-743f-4b3e-80ee-adb520920281',
            emailAddress: 'lroling-384934@j.mail',
            firstName: 'Luke',
            lastName: 'Roling',
            primaryAddressType: 'WORK'
          },
          attendeeId: 'aac17ae9-74b8-4687-ae9e-e2781d247f95'
        },
        attendeeType: 'ATTENDEE',
        productRegistrations: [
          {
            productId: Object.keys(EventSnapshot.eventSnapshot.products.admissionItems)[0],
            productType: 'AdmissionItem',
            quantity: 1,
            requestedAction: 'REGISTER'
          }
        ],
        quantityItemRegistrations: {},
        sessionRegistrations: {
          [unSelectedSessionId]: {
            productId: unSelectedSessionId,
            requestedAction: 'REGISTER',
            registrationSourceType: 'Selected'
          },
          limitedSelectedSession: {
            productId: 'limitedSelectedSession',
            requestedAction: 'REGISTER',
            registrationSourceType: 'Selected'
          }
        },
        sessionWaitlists: {
          [waitlistedSessionId]: {
            productId: waitlistedSessionId,
            requestedAction: 'WAITLIST',
            registrationSourceType: 'Selected'
          }
        },
        donationItemRegistrations: {
          donationItem2: {
            productId: 'donationItem2',
            amount: '7'
          }
        },
        registrationTypeId: '00000000-0000-0000-0000-000000000000',
        registrationPathId
      }
    }
  };
  return {
    website: EventSnapshot.eventSnapshot.siteEditor.website,
    accessToken,
    clients: {
      // @ts-expect-error ts-migrate(2350) FIXME: Only a void function can be called with the 'new' ... Remove this comment to see the full error message
      regCartClient: new RegCartClient()
    },
    registrationForm: {
      regCartPayment: {
        pricingInfo: {
          selectedPaymentMethod: null
        }
      },
      regCart,
      errors: {}
    },
    regCartStatus: {
      registrationIntent: REGISTERING,
      lastSavedRegCart: regCart
    },
    event: {
      eventFeatureSetup: {
        fees: {
          fees: false
        },
        registrationProcess: {
          multipleRegistrationTypes: false
        },
        agendaItems: {
          admissionItems: true
        }
      },
      products: {
        sessionContainer: {
          optionalSessions: {
            sessionId: {
              id: 'sessionId',
              capacityId: 'sessionId'
            },
            limitedSelectedSession: {
              id: 'limitedSelectedSession',
              capacityId: 'limitedSelectedSession'
            },
            limitedUnSelectedSession: {
              id: 'limitedUnSelectedSession',
              capacityId: 'limitedUnSelectedSession'
            },
            unSelectedSessionId: {
              id: unSelectedSessionId,
              capacityId: unSelectedSessionId
            }
          }
        }
      }
    },
    visibleProducts: {
      Sessions: {
        [primaryEventRegId]: { ...visibleProductsForRegistration }
      }
    },
    appData,
    text: {
      translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx),
      resolver: {
        date: () => 'some date',
        currency: x => x
      }
    },
    userSession: {
      regCartId: '928f02df-4ba7-4637-824d-ae71bd414e4f'
    },
    defaultUserSession: {},
    waitlistSelectionForGuests: {
      [waitlistedSessionId]: true
    },
    capacity: {
      ...dummyCapacitySummaries
    },
    experiments: {}
  };
}

export const response = {
  // mock response with getters to always use fresh copy
  get regCart(): $TSFixMe {
    return {
      regCartId,
      status: 'INPROGRESS',
      eventRegistrations: {
        [primaryEventRegId]: {
          eventRegistrationId: primaryEventRegId,
          attendee: {
            personalInformation: {
              emailAddress: 'lroling-384934@j.mail'
            },
            attendeeId: 'aac17ae9-74b8-4687-ae9e-e2781d247f95'
          },
          productRegistrations: [
            {
              productId: Object.keys(EventSnapshot.eventSnapshot.products.admissionItems)[0],
              productType: 'AdmissionItem',
              quantity: 1,
              requestedAction: 'REGISTER'
            }
          ],
          sessionRegistrations: {},
          registrationTypeId: '00000000-0000-0000-0000-000000000000',
          registrationPathId: '411c6566-1e5a-4c38-b8e5-f63ab9239b40'
        }
      }
    };
  }
};
