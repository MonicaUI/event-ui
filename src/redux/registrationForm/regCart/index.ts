import { lazyLoadAction, lazyLoadFunction } from '../../../utils/lazyLoad';

export const updateSessionRegistration = lazyLoadFunction(() =>
  import(/* webpackChunkName: "registration" */ './internal').then(m => m.updateSessionRegistration)
);

export const updateAdmissionItemRegistration = lazyLoadFunction(() =>
  import(/* webpackChunkName: "registration" */ './internal').then(m => m.updateAdmissionItemRegistration)
);

export const getPricing = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './pricing').then(m => (m as $TSFixMe).getPricing)
);

export const updateRegCartWithGuests = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateRegCartWithGuests)
);

export const updateGuestsInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateGuestsInRegCart)
);

export const setCurrentGuestEventRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.setCurrentGuestEventRegistration)
);

export const removeGuestByEventRegistrationId = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.removeGuestByEventRegistrationId)
);

export const updateGuestDetails = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateGuestDetails)
);

export const clearTemporaryGuestInformation = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.clearTemporaryGuestInformation)
);

export const removeEventRegistrationFromRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.removeEventRegistrationFromRegCart)
);

export const removeGroupMembersFromRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.removeGroupMembersFromRegCart)
);

export const addGroupMemberInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.addGroupMemberInRegCart)
);

export const navigateToGroupMemberRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.navigateToGroupMemberRegistration)
);

export const startRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startRegistration)
);

export const abortRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.abortRegCart)
);

export const abortRegCartAndLogout = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.abortRegCartAndLogout)
);

export const rescindAbortRegCartAndLogoutRequest = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.rescindAbortRegCartAndLogoutRequest)
);

export const startModification = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startModification)
);

export const finalizeRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeRegistration)
);

export const calculateServiceFeesForPartialPayments = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.calculateServiceFeesForPartialPayments)
);

export const resumeAlreadyStartedCheckout = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.resumeAlreadyStartedCheckout)
);

export const waitForRegCartCheckoutCompletionUi = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.waitForRegCartCheckoutCompletionUi)
);

export const startCancelRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startCancelRegistration)
);

export const finalizeCancelRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeCancelRegistration)
);

export const startDeclineRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startDeclineRegistration)
);

export const startWaitlistRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startWaitlistRegistration)
);

export const finalizeWaitlistRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeWaitlistRegistration)
);

export const finalizeDeclineRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeDeclineRegistration)
);

export const saveRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.saveRegistration),
  true
);

export const updateRegCart = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.updateRegCart)
);

export const updatePaymentCreditsInRegCart = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './paymentCredits').then(m => m.updatePaymentCreditsInRegCart)
);

export const updateTravelFlagInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.updateTravelFlagInRegCart)
);

export const setAirRequestOptOutChoice = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.setAirRequestOptOutChoice)
);

export const applyPartialEventRegistrationUpdate = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () =>
    import(/* webpackChunkName: "registration" */ './partialUpdates').then(m => m.applyPartialEventRegistrationUpdate)
);

export const setRegistrationTypeId = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './registrationTypes').then(m => m.setRegistrationTypeId)
);

export const runRegistrationTypeChangeValidationsForPrimaryAndGuest = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () =>
    import(/* webpackChunkName: "registration" */ './registrationTypes').then(
      m => m.runRegistrationTypeChangeValidationsForPrimaryAndGuest
    )
);

export const restoreRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.restoreRegistration)
);

export const restoreRegistrationFromOtherTab = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.restoreRegistrationFromOtherTab)
);

export const resumePartialRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.resumePartialRegistration)
);

export const searchPartialRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.searchPartialRegistration)
);
