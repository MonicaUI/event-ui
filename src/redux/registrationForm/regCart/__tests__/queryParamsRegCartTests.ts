import { startRegistration } from '../index';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import EventSnapshot from '../../../../../fixtures/EventSnapshot.json';
// eslint-disable-next-line jest/no-mocks-import
import { getState, response } from '../__mocks__/regCart';

function RegCartClient() {}

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const eventId = EventSnapshot.eventSnapshot.id;

jest.mock('querystring', () => {
  // to effectively set window.location.search
  const querystring = jest.requireActual<$TSFixMe>('querystring');
  return {
    ...querystring,
    parse: querystring.parse.bind(null, 'MarketoID=marketo-id&c=contact-id&contactWebsite=true')
  };
});

jest.mock('../../../actions', () => {
  return {
    ...jest.requireActual<$TSFixMe>('../../../actions'),
    filterEventSnapshot: () => () => {},
    loadRegistrationContent: () => () => {}
  };
});

RegCartClient.prototype.createRegCart = jest.fn(() => ({ regCart: response.regCart }));

afterEach(() => {
  jest.clearAllMocks();
});

test('creates a regCart with external reg contact id', async () => {
  const store = mockStore(getState());
  store.experiments = {
    ...store.experiments
  };
  await store.dispatch(startRegistration({ eventId }));
  const regCartClient = store.getState().clients.regCartClient;
  expect(regCartClient.createRegCart).toHaveBeenCalledTimes(1);
  expect(store.getActions()).toMatchSnapshot();
});

test('creates a regCart with external reg contact id - cached regCart pricing', async () => {
  const store = mockStore(getState());
  store.experiments = {
    ...store.experiments
  };
  await store.dispatch(startRegistration({ eventId }));
  const regCartClient = store.getState().clients.regCartClient;
  expect(regCartClient.createRegCart).toHaveBeenCalledTimes(1);
  expect(store.getActions()).toMatchSnapshot();
});
