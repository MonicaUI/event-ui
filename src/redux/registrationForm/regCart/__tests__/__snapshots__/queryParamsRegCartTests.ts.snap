// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a regCart with external reg contact id - cached regCart pricing 1`] = `
Array [
  Object {
    "type": "nucleus-guestside-site/loading/SHOW_LOADING_DIALOG",
  },
  Object {
    "payload": Object {
      "referenceId": undefined,
      "regCartId": "928f02df-4ba7-4637-824d-ae71bd414e4f",
    },
    "type": "event-guestside-site/registrantLogin/UPDATE_USER_SESSION",
  },
  Object {
    "type": "event-guestside-site/regCart/CREATE_REG_CART_PENDING",
  },
  Object {
    "payload": Object {
      "capacity": Object {
        "93e7463b-ebaa-40e7-b0ad-a2ca26f87b3c": Object {
          "active": true,
          "availableCapacity": -1,
          "capacityId": "93e7463b-ebaa-40e7-b0ad-a2ca26f87b3c",
          "totalCapacityAvailable": -1,
        },
        "c0215717-5640-4e9d-b790-36047f14bf21": Object {
          "active": true,
          "availableCapacity": -1,
          "capacityId": "c0215717-5640-4e9d-b790-36047f14bf21",
          "totalCapacityAvailable": -1,
        },
      },
    },
    "type": "event-widgets/capacity/LOAD_CAPACITY",
  },
  Object {
    "payload": Object {
      "currentEventRegistrationId": "00000000-0000-0000-0000-000000000001",
      "regCart": Object {
        "eventRegistrations": Object {
          "00000000-0000-0000-0000-000000000001": Object {
            "attendee": Object {
              "attendeeId": "aac17ae9-74b8-4687-ae9e-e2781d247f95",
              "personalInformation": Object {
                "contactId": "d10d9eab-743f-4b3e-80ee-adb520920281",
                "emailAddress": "lroling-384934@j.mail",
                "firstName": "Luke",
                "lastName": "Roling",
                "primaryAddressType": "WORK",
              },
            },
            "attendeeType": "ATTENDEE",
            "eventId": "dd48b4c4-672e-431c-be8d-5abfec299844",
            "eventRegistrationId": "00000000-0000-0000-0000-000000000001",
            "productRegistrations": Array [
              Object {
                "productId": "c0215717-5640-4e9d-b790-36047f14bf21",
                "productType": "AdmissionItem",
                "quantity": 1,
                "requestedAction": "REGISTER",
              },
            ],
            "registrationPathId": "411c6566-1e5a-4c38-b8e5-f63ab9239b40",
            "registrationTypeId": "00000000-0000-0000-0000-000000000000",
            "sessionRegistrations": Object {},
          },
        },
        "eventSnapshotVersions": Object {
          "undefined": "fake-eventSnapshot-version",
        },
        "regCartId": "928f02df-4ba7-4637-824d-ae71bd414e4f",
        "status": "INPROGRESS",
      },
      "validationMessages": undefined,
    },
    "type": "event-guestside-site/regCart/CREATE_REG_CART_SUCCESS",
  },
  Object {
    "payload": Object {
      "eventRegistrationId": undefined,
      "visibleProducts": undefined,
    },
    "type": "event-guestside-site/REPLACE_VISIBLE_SESSION_PRODUCTS",
  },
  Object {
    "payload": Object {
      "questionIds": Array [],
      "visibility": Object {},
    },
    "type": "event-guestside-site/UPDATE_QUESTION_VISIBILITY",
  },
  Object {
    "type": "nucleus-guestside-site/loading/HIDE_LOADING_DIALOG",
  },
]
`;

exports[`creates a regCart with external reg contact id 1`] = `
Array [
  Object {
    "type": "nucleus-guestside-site/loading/SHOW_LOADING_DIALOG",
  },
  Object {
    "payload": Object {
      "referenceId": undefined,
      "regCartId": "928f02df-4ba7-4637-824d-ae71bd414e4f",
    },
    "type": "event-guestside-site/registrantLogin/UPDATE_USER_SESSION",
  },
  Object {
    "type": "event-guestside-site/regCart/CREATE_REG_CART_PENDING",
  },
  Object {
    "payload": Object {
      "capacity": Object {
        "93e7463b-ebaa-40e7-b0ad-a2ca26f87b3c": Object {
          "active": true,
          "availableCapacity": -1,
          "capacityId": "93e7463b-ebaa-40e7-b0ad-a2ca26f87b3c",
          "totalCapacityAvailable": -1,
        },
        "c0215717-5640-4e9d-b790-36047f14bf21": Object {
          "active": true,
          "availableCapacity": -1,
          "capacityId": "c0215717-5640-4e9d-b790-36047f14bf21",
          "totalCapacityAvailable": -1,
        },
      },
    },
    "type": "event-widgets/capacity/LOAD_CAPACITY",
  },
  Object {
    "payload": Object {
      "currentEventRegistrationId": "00000000-0000-0000-0000-000000000001",
      "regCart": Object {
        "eventRegistrations": Object {
          "00000000-0000-0000-0000-000000000001": Object {
            "attendee": Object {
              "attendeeId": "aac17ae9-74b8-4687-ae9e-e2781d247f95",
              "personalInformation": Object {
                "contactId": "d10d9eab-743f-4b3e-80ee-adb520920281",
                "emailAddress": "lroling-384934@j.mail",
                "firstName": "Luke",
                "lastName": "Roling",
                "primaryAddressType": "WORK",
              },
            },
            "attendeeType": "ATTENDEE",
            "eventId": "dd48b4c4-672e-431c-be8d-5abfec299844",
            "eventRegistrationId": "00000000-0000-0000-0000-000000000001",
            "productRegistrations": Array [
              Object {
                "productId": "c0215717-5640-4e9d-b790-36047f14bf21",
                "productType": "AdmissionItem",
                "quantity": 1,
                "requestedAction": "REGISTER",
              },
            ],
            "registrationPathId": "411c6566-1e5a-4c38-b8e5-f63ab9239b40",
            "registrationTypeId": "00000000-0000-0000-0000-000000000000",
            "sessionRegistrations": Object {},
          },
        },
        "eventSnapshotVersions": Object {
          "undefined": "fake-eventSnapshot-version",
        },
        "regCartId": "928f02df-4ba7-4637-824d-ae71bd414e4f",
        "status": "INPROGRESS",
      },
      "validationMessages": undefined,
    },
    "type": "event-guestside-site/regCart/CREATE_REG_CART_SUCCESS",
  },
  Object {
    "payload": Object {
      "eventRegistrationId": undefined,
      "visibleProducts": undefined,
    },
    "type": "event-guestside-site/REPLACE_VISIBLE_SESSION_PRODUCTS",
  },
  Object {
    "payload": Object {
      "questionIds": Array [],
      "visibility": Object {},
    },
    "type": "event-guestside-site/UPDATE_QUESTION_VISIBILITY",
  },
  Object {
    "type": "nucleus-guestside-site/loading/HIDE_LOADING_DIALOG",
  },
]
`;
